docs/source/bmi.extensions.md

```diff
@@ -0,0 +1,63 @@
+(extensions)=
+
+# BMI Extensions
+
+An extension of BMI is identified by a unique name string. It could
+encompass one or several of the following:
+
+- Added functions exposed through extension objects that models export
+  via `get-extension-object`
+- Variable sets that callers can expect models to publish
+- Shared conventions of calling sequences or protocols around the use
+  of existing functions in core BMI or other extensions
+
+Models implementing BMI are not required to support these extensions,
+nor any others.
+
+
+## Added Functions
+
+Added functions in an extension must be expressed as members of an
+{term}`extension object`. Models expose these extension objects via
+`get-extension-object`. The name string should match the name of the
+extension itself. If an extension needs to expose multiple distinct
+objects, the name used for each should be an elaboration of the
+extension's name.
+
+The extension must specify the type signature of each extension object
+it defines, with all of its members. For interoperability, develoeprs
+are encouraged to use types and calling conventions that follow the
+practices of core BMI. However, this standard also anticipates that
```

Comment:
Indexed/numbered variables could help alleviate string ABI interoperability in implementing extensions in various languages.